### ISSUE TITLE
Number of classes

### ISSUE THREAD

**Issue by Roopg** [qubvel/segmentation_models#488]
Hi 

I have a query about the number of classes, as per this repo, for binary number of classes should be 1, but as per solutions discussed in issue #233 , number of classes should be 2 for proper training of the network. This is a bit confusing.
Also I do not see one hot encoding in the original repo by qubvel, so wondering how proper segmentation results were achieved.
I encoded my binary mask and expanded dimensions as required, but I had to get rid of the following code from this repo-

    #extract certain classes from mask (e.g. cars)
       masks = [(mask == v) for v in self.class_values]
       mask = np.stack(masks, axis=-1).astype('float')
    # # add background if mask is not binary
        # if mask.shape[-1] != 1:
        #     background = 1 - mask.sum(axis=-1, keepdims=True)
        #     mask = np.concatenate((mask, background), axis=-1)
        #     print(mask.shape)
so now the shape of my mask is : (1, 480, 320, 2) and image is (1, 480,320,3) 

But now I am running into the following issues-

1) Binary mask is inverted: 1 (white) for background and 0(black) for the lesion(class of interest).

2) Error in the generation of the training augmentation batch:
-->  img = cv2.warpAffine(img, matrix, (width, height), flags=interpolation, borderMode=border_mode, borderValue=value)
       return img
error: OpenCV(4.5.3) /tmp/pip-req-build-fvfwe_ss/opencv/modules/imgproc/src/imgwarp.cpp:2595: error: (-215:Assertion failed) src.cols > 0 && src.rows > 0 in function 'warpAffine'

In other words, what specifically needs to be changed in this "binary example repo"  to handle the binary segmentation case where there is only one class of interest (for eg tumor etc) and the other is the background.

**Comment by johnfoxy** [comment]
Did you manage to sort this out? If so, please post your code?

**Comment by Roopg** [comment]
Yes I did. Thank you @johnfoxy
